src/serac/physics/utilities/functional/boundary_integral.hpp

```diff
@@ -340,6 +387,14 @@ class BoundaryIntegral {
    */
   void GradientMult(const mfem::Vector& input_E, mfem::Vector& output_E) const { gradient_(input_E, output_E); }
 
+  /**
+   * @brief Computes the element stiffness matrices, storing them in an `mfem::Vector` that has been reshaped into a
+   * multidimensional array
+   * @param[inout] K_b The reshaped vector as a mfem::DeviceTensor of size (test_dim * test_dof, trial_dim * trial_dof,
+   * elem)
+   */
+  void ComputeElementMatrices(mfem::Vector& K_b) const { gradient_mat_(K_b); }
```

Comment:
What is the use case for this being a public method?

Reply:
I think in either case the `mfem::Vector` itself doesn't really contain enough information to give the "Matrices" or "Gradients" meaning.

Ideally in the future I think we should move away from returning reshaped vector since it could easily be reshaped incorrectly back into the element matrices. I'd prefer the result to be a "Matrix format" and something like a vector for the QoI case.

Reply:
I agree that mfem::Vector is clearly the wrong container for this, but I think we stick to it for the near future until the axom array stuff is ready. I'm renaming things these quantities from "Matrix" to "Gradient" where appropriate.